tests/integration/test_core.py

```diff
@@ -191,3 +191,66 @@ async def test_wordpress_theme_installation_error(
         assert (
                 unit.workload_status == ops.model.ActiveStatus.name
         ), "status should back to active after invalid theme removed from config"
+
+@pytest.mark.asyncio
+async def test_wordpress_install_uninstall_plugins(
+        ops_test: pytest_operator.plugin.OpsTest,
+        application_name,
+        unit_ip_list,
+        get_plugin_list_from_ip
+):
+    """
+    arrange: after WordPress charm has been deployed and db relation established
+    act: change plugins setting in config
+    assert: plugins should be installed and uninstalled accordingly
+    """
+    plugin_change_list = [
+        ["classic-editor", "classic-widgets"],
+        ["classic-editor"],
+        ["classic-widgets"],
+        []
+    ]
+    for plugins in plugin_change_list:
+        application = ops_test.model.applications[application_name]
+        await application.set_config({"plugins": ",".join(plugins)})
+        await ops_test.model.wait_for_idle()
+
+        for unit_ip in unit_ip_list:
+            expected_plugins = set(plugins)
+            expected_plugins.update(WordpressCharm._WORDPRESS_DEFAULT_PLUGINS)
+            assert (
+                    expected_plugins == set(get_plugin_list_from_ip(unit_ip))
+            ), f"plugin installed {plugins} should match plugins setting in config"
+
+
+@pytest.mark.asyncio
+async def test_wordpress_plugin_installation_error(
+        ops_test: pytest_operator.plugin.OpsTest,
+        application_name
+):
+    """
+    arrange: after WordPress charm has been deployed and db relation established
+    act: install a nonexistent plugin
+    assert: charm should switch to blocked state and the reason should be included in the status
+    message.
```

Comment:
Could this be indented by 4 spaces to indicate that it is still part of assert (similar to how long parameter descriptions etc work in docstrings)? It is a shortcoming of the standard not to define the formatting of long messages

Reply:
Fixed, thanks!